.builder/actions/mock_server_setup.py

```diff
@@ -0,0 +1,40 @@
+"""
+Setup local mock server for tests
+"""
+
+import Builder
+
+import os
+import subprocess
+
+
+class MockServerSetup(Builder.Action):
+    """
+    Set up this machine for running the mock server test
+
+    This action should be run in the 'pre_build_steps' or 'build_steps' stage.
+    """
+
+    def run(self, env):
+        self.env = env
+        if os.name == 'nt':
+            python_execute = "python"
+        else:
+            python_execute = "python3"
+
+        # install dependency for mock server
+        result = self.env.shell.exec(python_execute,
+                                     '-m', 'install', 'h11', 'trio')
+        if result.returncode != 0:
+            print("Mock server failed to setup, skip the mock server tests.")
+            return
```

Comment:
I am not sure which is better, error out or return..

Reply:
Is it failing to install anywhere? If it works everywhere, then make it an error

If we just return: there's a chance that setup starts failing everywhere, but we never realize because it's a silent warning

If we make it an error: then you can end up spending waaaaay too much time trying to get it working on esoteric/old platforms like manylinux1.

Reply:
current it's not failing from anywhere. But, two weeks ago, I think github ubuntu 20 image was failing to install trio for "I don't know" reason. Let's make it fail and heck around the image that doesn't support it.